components/kots-config-check/storage/entrypoint.sh

```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+# Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+# Licensed under the GNU Affero General Public License (AGPL).
+# See License-AGPL.txt in the project root for license information.
+
+set -euo pipefail
+
+STORE_PROVIDER="${1:-""}"
+STORE_LOCATION="${2:-""}"
+AZURE_ACCOUNT_NAME="${3:-""}"
+AZURE_ACCESS_KEY="${4:-""}"
+GCP_PROJECT_ID="${5:-""}"
+GCP_SERVICE_ACCOUNT_KEY="${6:-""}"
+S3_ENDPOINT="${7:-""}"
+S3_ACCESS_KEY_ID="${8:-""}"
+S3_SECRET_ACCESS_KEY="${9:-""}"
+
+bucket_name="kots-check-${RANDOM}-${RANDOM}"
+downloaded_file=/tmp/download
+file_name="kots-check-file"
+file_contents="$(date)"
+uploaded_file=/tmp/upload
+
+echo "${file_contents}" > "${uploaded_file}"
+
+connection="false"
+
+function test_azure() {
+  echo "Using Azure storage"
+
+  echo "Create a container"
+  az storage container create \
+    --account-name "${AZURE_ACCOUNT_NAME}" \
+    --account-key "${AZURE_ACCESS_KEY}" \
+    --name "${bucket_name}" || return 1
+
+  echo "Upload a file"
+  az storage blob upload \
+    --account-name "${AZURE_ACCOUNT_NAME}" \
+    --account-key "${AZURE_ACCESS_KEY}" \
+    --container-name "${bucket_name}" \
+    --file "${uploaded_file}" \
+    --name "${file_name}" || return 1
+
+  echo "Download the file"
+  az storage blob download \
+    --account-name "${AZURE_ACCOUNT_NAME}" \
+    --account-key "${AZURE_ACCESS_KEY}" \
+    --container-name "${bucket_name}" \
+    --file "${downloaded_file}" \
+    --name "${file_name}" || return 1
+
+  echo "Compare the file"
+  diff "${downloaded_file}" "${uploaded_file}" || return 1
+
+  echo "Delete the container"
+  az storage container delete \
+    --name "${bucket_name}" \
+    --account-name "${AZURE_ACCOUNT_NAME}" \
+    --account-key "${AZURE_ACCESS_KEY}" || return 1
+}
+
+function test_gcp() {
+  echo "Using GCP storage"
+
+  echo "${GCP_SERVICE_ACCOUNT_KEY}" | base64 -d > /tmp/creds.json
+
+  gcloud auth activate-service-account --project="${GCP_PROJECT_ID}" --key-file=/tmp/creds.json
+
+  echo "Create bucket"
+  gsutil mb \
+    -l "${STORE_LOCATION}" \
+    "gs://${bucket_name}" || return 1
+
+  echo "Upload a file"
+  gsutil cp \
+    "${uploaded_file}" \
+    "gs://${bucket_name}/${file_name}" || return 1
+
+  echo "Download a file"
+  gsutil cp \
+    "gs://${bucket_name}/${file_name}" \
+    "${downloaded_file}" || return 1
+
+  echo "Compare the file"
+  diff "${downloaded_file}" "${uploaded_file}" || return 1
+
+  echo "Delete bucket"
+  gsutil rm -r \
+    "gs://${bucket_name}" || return 1
+}
+
+function test_s3() {
+  echo "Using S3 storage"
+
+  mc alias set s3 "https://${S3_ENDPOINT}" "${S3_ACCESS_KEY_ID}" "${S3_SECRET_ACCESS_KEY}"
+
+  echo "Create bucket"
+  mc mb \
+    --region="${STORE_LOCATION}" \
+    "s3/${bucket_name}" || return 1
+
+  echo "Upload a file"
+  mc cp \
+    "${uploaded_file}" \
+    "s3/${bucket_name}/${file_name}" || return 1
+
+  echo "Download a file"
+  mc cp \
+    "s3/${bucket_name}/${file_name}" \
```

Comment:
We need to consider the static bucket name introduced in https://github.com/gitpod-io/gitpod/pull/10188. This is a blocker. :disappointed: